syncer/syncer.go

```diff
@@ -184,6 +186,7 @@ func NewSyncer(cfg *config.SubTaskConfig, etcdClient *clientv3.Client) *Syncer {
 	logger := log.With(zap.String("task", cfg.Name), zap.String("unit", "binlog replication"))
 	syncer := &Syncer{
 		pessimist: shardddl.NewPessimist(&logger, etcdClient, cfg.Name, cfg.SourceID),
+		optimist:  shardddl.NewOptimist(&logger, etcdClient, cfg.Name, cfg.SourceID),
```

Comment:
Shall we start just `pessimist/optimist` here according to the `cfg.ShardMode`? But it doesn't matter a lot, we can change this in the future.

Reply:
We may need to support `switch-shard-mode` dynamically later 😭